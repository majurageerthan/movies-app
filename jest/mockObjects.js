export const movie = {
  adult: false,
  backdrop_path: '/qdIMHd4sEfJSckfVJfKQvisL02a.jpg',
  genre_ids: [
    35,
    18,
    10749,
  ],
  id: 13,
  original_language: 'en',
  original_title: 'Forrest Gump',
  // eslint-disable-next-line max-len
  overview: 'A man with a low IQ has accomplished great things in his life and been present during significant historic eventsâ€”in each case, far exceeding what anyone imagined he could do. But despite all he has achieved, his one true love eludes him.',
  popularity: 68.355,
  poster_path: '/saHP97rTPS5eLmrLQEcANmKrsFl.jpg',
  release_date: '1994-06-23',
  title: 'Forrest Gump',
  video: false,
  vote_average: 8.5,
  vote_count: 22910,
};

export const movies = [movie, movie];
