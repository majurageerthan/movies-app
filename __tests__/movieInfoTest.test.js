import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfoText from '../src/components/molecules/movieInfoText';
import { movie } from '../jest/mockObjects';

// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  const tree = renderer.create(<MovieInfoText
    movie={movie}
    index={1}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
