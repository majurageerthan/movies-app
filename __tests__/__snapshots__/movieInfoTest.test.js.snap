// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    #2 Forrest Gump
  </Text>
  <Text
    style={
      Object {
        "color": "#808080",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    (Forrest Gump)
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 7,
        "marginTop": 5,
        "maxHeight": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#808080",
        }
      }
    >
      1994-06-23
    </Text>
    <Text
      style={
        Object {
          "color": "#808080",
          "paddingRight": 1,
        }
      }
    >
       (en) . 
    </Text>
    <View
      style={
        Object {
          "borderColor": "#D3D3D3",
          "borderWidth": 1,
          "justifyContent": "center",
          "padding": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#808080",
            "fontSize": 12,
          }
        }
      >
        All
      </Text>
    </View>
  </View>
  <Text
    numberOfLines={4}
    style={
      Object {
        "color": "#808080",
      }
    }
  >
    A man with a low IQ has accomplished great things in his life and been present during significant historic eventsâ€”in each case, far exceeding what anyone imagined he could do. But despite all he has achieved, his one true love eludes him.
  </Text>
</View>
`;
